tfhe/src/integer/server_key/radix_parallel/scalar_comparison.rs

```diff
@@ -881,17 +881,17 @@ impl ServerKey {
         let block_states = {
             for i in 1..grouping_size {
                 let state_fn = |block| {
-                    let r = (u64::MAX * u64::from(block != 0)) % (packed_modulus * 2);
-                    r << (i - 1)
+                    let r = u64::MAX * u64::from(block != 0);
+                    (r << (i - 1)) % (packed_modulus * 2)
```

Comment:
this is not stricticly equivalent here is it ?

Reply:
The new version is the more correct one, as the modulus needs to be applied after

Reply:
same question is there a way to detect this with a test ?

Reply:
and pinging the GPU

Reply:
ok